src/it/java/org/eclipse/microprofile/starter/utils/Commands.java

```diff
@@ -0,0 +1,265 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import static org.eclipse.microprofile.starter.TestMatrixTest.API_URL;
+import static org.eclipse.microprofile.starter.TestMatrixTest.TMP;
+import static org.eclipse.microprofile.starter.utils.Logs.S;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author Michal Karm Babacek <karm@redhat.com>
+ */
+public class Commands {
+    private static final Logger LOGGER = Logger.getLogger(Commands.class.getName());
+
+    private static final String STARTER_TS_WORKSPACE = "STARTER_TS_WORKSPACE";
+
+    public static String getWorkspaceDir() {
+        String env = System.getenv().get(STARTER_TS_WORKSPACE);
+        String sys = System.getProperty(STARTER_TS_WORKSPACE);
+        String fallback = System.getProperty("java.io.tmpdir");
+        if (StringUtils.isNotBlank(env)) {
+            return env;
+        }
+        if (StringUtils.isNotBlank(sys)) {
+            return sys;
+        }
+        return fallback;
+    }
+
+    public static void download(Client client, String supportedServer, String artifactId, SpecSelection specSelection, String location) {
+        Response response = client.target(API_URL + "/project?supportedServer=" + supportedServer + specSelection.queryParam + "&artifactId=" + artifactId).request().get();
+        assertEquals("Download failed.", Response.Status.OK.getStatusCode(), response.getStatus());
+        try (FileOutputStream out = new FileOutputStream(location); InputStream in = (InputStream) response.getEntity()) {
+            in.transferTo(out);
+            out.flush();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static File unzip(String location, String artifactId) throws InterruptedException, IOException {
+        ProcessBuilder pb;
+        if (isThisWindows()) {
+            pb = new ProcessBuilder("powershell", "-c", "Expand-Archive", "-Path", location, "-DestinationPath", TMP, "-Force");
+        } else {
+            pb = new ProcessBuilder("unzip", "-o", location, "-d", TMP);
+        }
+        Map<String, String> env = pb.environment();
+        env.put("PATH", System.getenv("PATH"));
+        pb.directory(new File(TMP));
+        File unzipLog = new File(TMP + S + artifactId + "-unzip.log");
+        pb.redirectErrorStream(true);
+        pb.redirectOutput(ProcessBuilder.Redirect.to(unzipLog));
+        Process p = pb.start();
+        // On slow cloud VMs with weird I/O, this could be minutes for some reason...
+        p.waitFor(3, TimeUnit.MINUTES);
+        return unzipLog;
+    }
+
+    public static void cleanWorkspace(String artifactId) {
+        String path = TMP + S + artifactId;
+        try {
+            FileUtils.deleteDirectory(new File(path));
+        } catch (IOException e) {
+            // Silence is golden
+        }
+        (new File(path + ".zip")).deleteOnExit();
```

Comment:
why `deleteOnExit` when you explicitly delete it in the next statements?

Reply:
"Deletion will be attempted only for normal termination of the virtual machine"

The reason is that if a file lock is held on Windows, getting rid of the file on shutdown might be the only way to have it deleted.  This is Windows specific.  Adding comments.